### PR TITLE
avocado: Add another branch in test detection heuristics

### PR DESCRIPTION
**Pull request by lmr**
It occurred to us that we can tell apart a buggy avocado
test from a random buggy module since all avocado instrumented
tests will have an avocado import. Therefore, let's do a regexp
search on the module to see if we find an avocado import header.

Of course, this still has room for mistakes, but this is a
best effort enterprise anyway. Update the diagram with the
extra branch in the test resolution.

Signed-off-by: Lucas Meneghel Rodrigues <lmr@redhat.com>

### CODE REVIEW

**Comment by lmr** [comment]
@ruda, please take a look

**Comment by adereis** [comment]
LGTM (I was discussing this with lmr on IRC). But I won't merge it because I don't think my superficial review here on github is enough.

File: avocado/loader.py [hunk 1]
```diff
@@ -99,18 +100,34 @@ def _make_test(self, test_name, test_path, params, queue):
...
-                # best guess is that this is a buggy test.
-                test_class = test.BuggyTest
-                params['exception'] = details
+                # Module can't be imported and it's not an executable. Let's
+                # see if there's an avocado import into the test. Although
+                # not entirely reliable, we hope it'll be good enough.
+                likely_avocado_test = False
+                with open(test_path, 'r') as test_file_obj:
+                    test_contents = test_file_obj.read()
+                    patterns = ['from avocado.* import', 'import avocado.*']
```
Comment (ruda) [inline avocado/loader.py]: I would better use `^from...` and `^import....`, to restrict the case where this line is present inside a string, and this is not a avocado test. Most of the time, for a test, we use imports in the column 0.

Comment (lmr) [inline avocado/loader.py]: OK

Comment (lmr) [inline avocado/loader.py]: I tried it and it doesn't match on a normal avocado test that had an intentional bug inserted on it.

**Comment by ruda** [comment]
Now it LGTM.

**Pull request merged**